src/main/scala/jtag/JtagStateMachine.scala

```diff
@@ -76,12 +74,8 @@ class JtagStateMachine(implicit val p: Parameters) extends Module() {
   }
   val io = IO(new StateMachineIO)
 
-  val nextState = WireInit(JtagState.TestLogicReset.U)
-  val currStateReg = Module (new AsyncResetRegVec(w = JtagState.State.width,
-    init = JtagState.State.toInt(JtagState.TestLogicReset)))
-  currStateReg.io.en := true.B
-  currStateReg.io.d  := nextState
-  val currState = currStateReg.io.q
+  val nextState = WireInit(JtagState.TestLogicReset.U)  // (JtagState.State.width.W))
```

Comment:
Are you concerned the width may not be properly inferred?

Reply:
It should be : https://github.com/chipsalliance/rocket-chip/blob/master/src/main/scala/jtag/JtagStateMachine.scala#L13

Reply:
unnecessary comment removed